src/backend/cgxmm.c

```diff
@@ -1344,5 +1348,58 @@ void checkSetVex3(code *c)
     }
 }
 
+/*************************************
+ * Determine if operation should be rewritten as a VEX
+ * operation; and do so.
+ * Params:
+ *      c = code
+ *      ty = type of operand
+ */
+
+void checkSetVex(code *c, tym_t ty)
+{
+    if (config.avx || tysize(ty) == 32)
```

Comment:
my suicide note will read "I've seen too many `if` statements that spanned entire functions instead of returning early".